text/0000-decouple-dependencies.md

```diff
@@ -0,0 +1,403 @@
+because the metadata file will be TOML format. For example, with
+`com.example.dep-a`, there would be the following folder structure:
+
+```
+com
+└── example
+    └── dep-a.toml
+```
+
+As mentioned previously, each individual metadata file has a file name
+consisting of the dependency name with an extension of `toml`. The internal
+format consists of a single table called `versions` which is an array of tables
+containing all of the versions for that particular dependency. Each version
+entry requires a `uri`, `version`, `checksum`, `arch`, `os`, and `license`. It
+may also have `name`, `purl`, `strip-components`, and `cpes` although these are
+optional. The `cpes` entry is an array of strings identifying all of the CPEs
+for that dependency. The `license` is itself an array of tables containing
+`type` and `uri` of the license for the dependency.
+
+For example:
+```toml
+[[versions]]
+cpes = [ "cpe:2.3:a:apache:maven:3.8.6:*:*:*:*:*:*:*" ]
+name = "Apache Maven"
+purl = "pkg:generic/apache-maven@3.8.6"
+checksum = "sha256:c7047a48deb626abf26f71ab3643d296db9b1e67f1faa7d988637deac876b5a9"
+arch = "x86_64"
+os = "linux"
+distro = "ubuntu-18.04"
+uri = "https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.8.6/apache-maven-3.8.6-bin.tar.gz"
+version = "3.8.6"
+strip-components = 1
+
+  [[versions.licenses]]
+  type = "Apache-2.0"
+  uri = "https://www.apache.org/licenses/"
+
+[[versions]]
+cpes = [ "cpe:2.3:a:apache:mvnd:0.7.1:*:*:*:*:*:*:*" ]
+name = "Apache Maven Daemon"
+purl = "pkg:generic/apache-mvnd@0.7.1"
+chekcsum = "sha256:ac0b276d4d7472d042ddaf3ad46170e5fcb9350981af91af6c5c13e602a07393"
+arch = "x86_64"
+os = "linux"
+uri = "https://github.com/apache/maven-mvnd/releases/download/0.7.1/mvnd-0.7.1-linux-amd64.zip"
+version = "0.7.1"
+
+  [[versions.licenses]]
+  type = "Apache-2.0"
+  uri = "https://www.apache.org/licenses/"
+```
+## Dependency Validation
+In `buildpack.toml` we will add a section to metadata for specifying dependency
+validation parameters. This is a way that buildpacks can state that they do or
+do not support certain versions of dependencies.
+
+A buildpack does not need to include this section, it is optional. If included,
+the buildpack and libraries like `libpak` and `packit` may use the information
+to fail if dependency versions are requested by a user that might cause
+problems for the buildpack. It is the buildpacks responsibility to process the
+validations and react to them, whether that be warning the user or even
+failing.
+
+The format for this metadata is such that you have an array of tables called
+`validations`. Each table in the array contains the dependency id, which
+follows the format of a dependency id as outlined in the [Metadata
+Format](#metadata-format) section. In addition, it includes an array of strings
+called `supported` which contains a list of [semver](https://semver.org/)
+ranges that indicate what is supported by that buildpack. Optionally it can
```

Comment:
So, for now, validations are only about versions? Probably an edge case, but the Java language family would benefit from having all Java vendors under a single namespace, wouldn't it? Has it been considered to add different validations in the future? If so, should this RFC state this somehow?

Reply:
I'm not as familiar with the Java ecosystem, but it seems to be that if you have something specific in mind you could propose it as an addition to this RFC, but otherwise it seems reasonable to defer that for a potential future RFC.

Reply:
So there is a discussion of sort of sub-team group domain `io.paketo.java.(dep)` for example that would allow us to group similar dependency together and create useful dependency metadata subset packages.

Reply:
My quest was if `io.paketo.java.*` would be about names for dependencies or for a separate packaging mechanism for dependencies...

I.e. I would expect the Paketo apache-tomcat buildpack to require a dependency named `org.apache.tomcat` rather than `io.paketo.java.tomcat`.